http3/client.go

```diff
@@ -83,6 +92,9 @@ func newClient(
 		opts:          opts,
 		dialer:        dialer,
 		logger:        logger,
+		clientContext: ctx,
+		clientCancel:  cancel,
+		clientStreams: &sync.Map{},
```

Comment:
You don't need any map here.

Reply:
I removed it as well, but i realized a (potential) issue. Can the client call RoundTrip for multiple different hosts and maintain those connections simultaneously?